### PR TITLE
feat(move-on-aptos): basic support for Move-on-Aptos

### PR DESCRIPTION
**Pull request by ArArgon**
For your information:
- Move-on-Aptos: https://aptos.dev/move/move-on-aptos
- Tree-sitter grammar: https://github.com/aptos-labs/tree-sitter-move-on-aptos
- `semgrep-move-on-aptos`: https://github.com/semgrep/semgrep-move-on-aptos
   Aptos' fork: https://github.com/aptos-labs/semgrep/tree/feat/move-on-aptos-integration

---

WIP:
- [x] Add test cases for move-on-aptos
- [x] Align with upstream `semgrep-move-on-aptos`:
    this PR will be ready once 1) more tests are done, 2) upstream PRs are merged.
- [ ] Semgrep JS? I don't know if this one is necessary for the integration

### CODE REVIEW

**Comment by aryx** [comment]
We can already merge this if you want @ArArgon . We can always improve upon in subsequent PRs.

**Comment by ArArgon** [comment]
> We can already merge this if you want @ArArgon . We can always improve upon in subsequent PRs.

We are okay with it. However, we need to merge PRs to `ocaml-tree-sitter-semgrep` and `semgrep-move-on-aptos` before merging this PR. This is because that for dev purpose, I linked the submodules to the feature branches. Now I should revert them one by one.

**Comment by ArArgon** [comment]
We are ready to merge! @aryx

**Review by aryx — APPROVED** [review]

**Pull request merged**
